src/main/scala/wdl4s/cwl/model.scala

```diff
@@ -0,0 +1,207 @@
+package wdl4s.cwl
+
+import eu.timepit.refined._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.string._
+import shapeless.{:+:, CNil}
+
+case class WorkflowStepInput(src: String)
+
+case class InputParameter(
+                           id: Option[String], //not really optional but can be specified upstream
+                           label: Option[String],
+                           secondaryFiles: Option[ECMAScriptExpression :+: String :+: Array[ECMAScriptExpression :+: String :+: CNil] :+: CNil],
+                           format: Option[ECMAScriptExpression :+: String :+: Array[String] :+: CNil],
+                           streamable: Option[Boolean],
+                           doc: Option[String :+: Array[String] :+: CNil],
+                           inputBinding: Option[CommandLineBinding],
+                           default: Option[String], //can be of type "Any" which... sucks.
```

Comment:
Sadly Any is not representable in Json, so a derivation of that type is not possible.  Honestly I'm surprised at such a loosely specified parameter.

What it means in reality is we have to figure out how to support some made-up `Any` type and write a custom parser that can take any of the JSON types.